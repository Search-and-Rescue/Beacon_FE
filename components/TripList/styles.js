const styles = {
  tripsListContainer: {
    alignItems: "center",
    height: "86%",
    marginTop: 20
  },
  modalToggleButton: {
    backgroundColor: "#EFB095",
    borderColor: "#001028",
    borderWidth: 4,
    marginHorizontal: "5%",
    marginTop: 10,
    paddingVertical: 10,
    width: "90%"
  },
  modalToggleText: {
    color: "#001028",
    fontFamily: "Futura",
    fontSize: 28,
    textAlign: "center"
  },
  tripsList: {
    backgroundColor: "fff",
    borderColor: "#001028",
    borderRadius: 3,
    borderWidth: 4,
    flex: 1,
    marginHorizontal: "5%",
    width: "90%"
  },
  backgroundImage: {
    flex: 1,
    width: "100%"
  },
  tripCard: {
    backgroundColor: "#efefef",
    flex: 1,
    flexDirection: "row",
    height: 75,
    marginVertical: 5,
    justifyContent: "space-between"
  },
  tripRemoveBtn: {
    color: "#001028",
    height: 30,
    marginHorizontal: 12,
    marginVertical: 20,
    padding: 5
  },
  tripTextContainer: {
    flex: 1,
    flexDirection: "column"
  },
  tripsName: {
    flex: 1,
    fontFamily: "Futura",
    fontSize: 21,
    paddingVertical: 2
  },
  tripsDate: {
    flex: 1,
    fontFamily: "Avenir",
    fontSize: 18,
  },
  modalView: {
    backgroundColor: "lightgrey",
    borderTopWidth: 10,
    borderColor: "#001028",
    margin: 5,
    padding: 20,
    position: "absolute",
    bottom: -5,
    left: 10,
    right: 10
  },
  theButton: {
    alignItems: "center",
    backgroundColor: "#EFB095",
    borderWidth: 10,
    borderColor: "#001028",
    color: "#EFB095",
    borderRadius: 60,
    paddingVertical: 100
  },
  theButtonText: {
    color: "#001028",
    fontFamily: "Futura",
    fontSize: 46
  },
  theButtonSubtext: {
    color: "#001028",
    fontSize: 28
  },
  remainActiveButton: {
    alignItems: "center",
    backgroundColor: "#001028",
    borderRadius: 3,
    marginVertical: 10,
    paddingVertical: 20
  },
  remainActiveButtonText: {
    color: "#F0F0F0",
    fontFamily: "Futura",
    fontSize: 28
  },
  addItemBtn: {
    borderColor: "black",
    borderWidth: 1,
    padding: 10,
    width: "100%"
  },
  addBtnText: {
    fontSize: 21,
    textAlign: "center"
  }
};

export default styles;
