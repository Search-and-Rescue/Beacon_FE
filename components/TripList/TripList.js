import React, { Component } from "react";
import { Button, ImageBackground, Modal, ScrollView, Text, TouchableOpacity, View } from "react-native";
// import { TouchableOpacity } from "react-native-gesture-handler";
import { deleteTrip, deactivateTrip, getTrips } from "../../util/apiCalls";
import { setTrips, removeCurrentTrip } from "../../actions";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import styles from "./styles";
import background from "../../assets/background.png";

export class TripList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      button_modal: false
    }
  }
  
  deactivateTripStatus = async id => {
    try {
      await deactivateTrip(id);
      this.props.removeCurrentTrip();
      this.toggleButtonModal();
    } catch ({ message }) {
      console.log(message);
    }
  } 

  removeTrip = async (id) => {
    if (id === this.props.currentTrip) {
      await deactivateTrip(id);
      this.props.removeCurrentTrip();
    }
    
    try {
      await deleteTrip(id);
      const userInfoTrips = await getTrips();
      this.props.setTrips(userInfoTrips.user.trips);
    } catch ({ message }) {
      console.log(message)
    }
  }

  toggleButtonModal = () => {
    this.setState({ button_modal: !this.state.button_modal });
  };

  render() {
    const tripCards = this.props.trips.map(trip => {
      return (
        <View key={trip.id} style={styles.tripCard}>
          <Text 
            style={styles.tripRemoveBtn}
            onPress={() => this.removeTrip(trip.id)}>REMOVE</Text>
          <View style={styles.tripTextContainer}>
            <Text style={styles.tripsName}>{trip.name}</Text>
            <Text style={styles.tripsDate}>
              {trip.startDate} to {trip.endDate}
            </Text>
          </View>
        </View>
      );
    });
    return (
      <View>
        {this.props.currentTrip && (
        <TouchableOpacity
          style={styles.modalToggleButton}
          onPress={() => this.toggleButtonModal()}
        >
          <Text style={styles.modalToggleText}>UPDATE TRIP STATUS</Text>
        </TouchableOpacity>
        )}
        {this.props.currentTrip && (
        <Modal
          animationType={"slide"}
          visible={this.state.button_modal}
          transparent={true}
          onRequestClose={() => console.log("close requested")}
        >
          <View style={styles.modalView}>
            <TouchableOpacity
              onPress={() => this.deactivateTripStatus(this.props.currentTrip)}
            >
              <View style={styles.theButton}>
                <Text style={styles.theButtonText}>DEACTIVATE</Text>
                <Text style={styles.theButtonText}>CURRENT TRIP</Text>
                <Text style={styles.theButtonSubtext}>Emergency notifications</Text>
                <Text style={styles.theButtonSubtext}>will not be sent.</Text>
              </View>
            </TouchableOpacity>
            <TouchableOpacity onPress={() => this.toggleButtonModal()}>
              <View style={styles.remainActiveButton}>
                <Text style={styles.remainActiveButtonText}>Remain Active</Text>
              </View>
            </TouchableOpacity>
          </View>
        </Modal>
       )}
        <View style={styles.tripsListContainer}>
          <ImageBackground
            source={background}
            style={styles.backgroundImage}
            imageStyle={{ opacity: 0.2 }}
          >
            <ScrollView style={styles.tripsList}>{tripCards}</ScrollView>
          </ImageBackground>
        </View>
      </View>
    );
  }
}

export const mapStateToProps = ({ trips, currentTrip }) => ({
  trips, 
  currentTrip
});

export const mapDispatchToProps = dispatch =>
  bindActionCreators({ setTrips, removeCurrentTrip }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(TripList);
