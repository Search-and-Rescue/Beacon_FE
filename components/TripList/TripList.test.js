import "react-native";
import React from "react";
import { shallow } from "enzyme";
import { TripList } from "./TripList";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

jest.mock("react-native-gesture-handler", () => "TouchableOpacity");npm 

describe("TripList", () => {
  let wrapper;
  const tripsMock = [{
    endDate: "2020-05-03",
    endTime: "2000-01-01 22:00:00 UTC",
    endingPoint: "Coyote Trail Head",
    name: "Spanish Peaks Backpacking Trip",
    notificationDate: "2020-05-04",
    notificationTime: "2000-01-01 15:00:00 UTC",
    startDate: "2020-05-01",
    startTime: "2000-01-01 08:00:00 UTC",
    startingPoint: "Coyote Trail Head",
    travelingCompanions: 0,
  }]

  beforeEach(() => {
    wrapper = shallow(<TripList
      trips={tripsMock} />)
  });

  it("should match the snapshot with all of the data passed through", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
