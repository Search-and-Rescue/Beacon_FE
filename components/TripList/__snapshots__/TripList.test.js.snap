// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TripList should match the snapshot with all of the data passed through 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": "96%",
        "marginTop": 20,
      }
    }
  >
    <ScrollViewMock
      style={
        Object {
          "borderColor": "black",
          "borderWidth": 2,
          "flex": 1,
          "height": 300,
          "width": "90%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#efefef",
            "borderColor": "black",
            "borderWidth": 1,
            "flex": 1,
            "flexDirection": "row",
            "height": 50,
            "justifyContent": "space-between",
            "marginVertical": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "borderColor": "black",
              "borderWidth": 1,
              "height": 35,
              "margin": 4,
              "padding": 5,
            }
          }
        >
          REMOVE
        </Text>
        <Text>
          Spanish Peaks Backpacking Trip
        </Text>
        <Text>
          2020-05-01
        </Text>
      </View>
    </ScrollViewMock>
  </View>
</View>
`;
